services/ymax-planner/test/kv-store-flow.test.ts

```diff
@@ -0,0 +1,214 @@
+/**
+ * Tests for the functionality of flows involving the KV store
+ */
+import test from 'ava';
+
+import { TxType } from '@aglocal/portfolio-contract/src/resolver/constants.js';
+import { createMockPendingTxData } from '@aglocal/portfolio-contract/tools/mocks.ts';
+import { getTxBlockLowerBound, setTxBlockLowerBound } from '../src/kv-store.ts';
+import { handlePendingTx } from '../src/pending-tx-manager.ts';
+import { EVENTS } from '../src/watchers/gmp-watcher.ts';
+import {
+  createMockGmpExecutionEvent,
+  createMockPendingTxOpts,
+} from './mocks.ts';
+
+const setupEnvironment = ({
+  latestBlock,
+  blockWithEvent,
+  txId,
+  contractAddress,
+}) => {
+  const logs: string[] = [];
+  const mockLog = (...args: unknown[]) => logs.push(args.join(' '));
+
+  const destinationAddress: `${string}:${string}:${string}` = `eip155:42161:${contractAddress}`;
+
+  const gmpTx = createMockPendingTxData({
+    type: TxType.GMP,
+    destinationAddress,
+  });
+
+  const chainId = 'eip155:42161';
+  const opts = createMockPendingTxOpts();
+  const mockProvider = opts.evmProviders[chainId] as any;
+
+  const currentTimeMs = 1700000000; // 2023-11-14T22:13:20Z
+  const txTimestampMs = currentTimeMs - 10 * 1000; // 10 seconds ago
+  const avgBlockTimeMs = 300; // 300 ms per block on eip155:42161
+  mockProvider.getBlockNumber = async () => latestBlock;
+
+  mockProvider.getBlock = async (blockNumber: number) => {
+    const blocksAgo = latestBlock - blockNumber;
+    const ts = currentTimeMs - blocksAgo * avgBlockTimeMs;
+    return { timestamp: Math.floor(ts / 1000) };
+  };
+
+  // Trigger block event to resolve waitForBlock
+  setTimeout(() => mockProvider.emit('block', latestBlock + 1), 10);
```

Comment:
It looks like this functionality actually belongs in `createMockPendingTxOpts` and `createMockProvider`.